In [15]:
import pandas as pd
import numpy as np
import math
import os
os.getcwd()


'C:\\Users\\Administrator\\Documents'

In [16]:
oj = pd.read_csv("C:\\Users\\Administrator\\Desktop\\Datasets\\python\\oj.csv")
oj



store      brand  week    logmove  feat  price     AGE60      EDUC  \
0          2  tropicana    40   9.018695     0   3.87  0.232865  0.248935   
1          2  tropicana    46   8.723231     0   3.87  0.232865  0.248935   
2          2  tropicana    47   8.253228     0   3.87  0.232865  0.248935   
3          2  tropicana    48   8.987197     0   3.87  0.232865  0.248935   
4          2  tropicana    50   9.093357     0   3.87  0.232865  0.248935   
5          2  tropicana    51   8.877382     0   3.87  0.232865  0.248935   
6          2  tropicana    52   9.294682     0   3.29  0.232865  0.248935   
7          2  tropicana    53   8.954674     0   3.29  0.232865  0.248935   
8          2  tropicana    54   9.049232     0   3.29  0.232865  0.248935   
9          2  tropicana    57   8.613230     0   3.29  0.232865  0.248935   
10         2  tropicana    58   8.680672     0   3.56  0.232865  0.248935   
11         2  tropicana    59   9.034080     0   3.56  0.232865  0.248935   
12         2  tropicana    60   8.691483     0   3.56  0.232865  0.248935   
13         2  tropicana    61   8.831712     0   3.56  0.232865  0.248935   
14         2  tropicana    62   9.128696     0   3.87  0.232865  0.248935   
15         2  tropicana    63   9.405907     0   2.99  0.232865  0.248935   
16         2  tropicana    64   9.447150     0   2.99  0.232865  0.248935   
17         2  tropicana    65   8.783856     0   3.59  0.232865  0.248935   
18         2  tropicana    66   8.723231     0   3.59  0.232865  0.248935   
19         2  tropicana    67   9.957976     0   2.39  0.232865  0.248935   
20         2  tropicana    68   9.426741     0   2.39  0.232865  0.248935   
21         2  tropicana    69   9.156095     0   3.59  0.232865  0.248935   
22         2  tropicana    70   9.793673     0   2.59  0.232865  0.248935   
23         2  tropicana    71   9.149316     0   2.59  0.232865  0.248935   
24         2  tropicana    72   8.743851     0   3.59  0.232865  0.248935   
25         2  tropicana    73   8.841014     0   3.59  0.232865  0.248935   
26         2  tropicana    74   9.727228     0   2.49  0.232865  0.248935   
27         2  tropicana    75   8.743851     0   3.59  0.232865  0.248935   
28         2  tropicana    76   8.979165     0   3.59  0.232865  0.248935   
29         2  tropicana    77   8.723231     0   3.59  0.232865  0.248935   
...      ...        ...   ...        ...   ...    ...       ...       ...   
28917    137  dominicks   131  10.404990     0   1.42  0.209602  0.528362   
28918    137  dominicks   132  10.715661     0   1.59  0.209602  0.528362   
28919    137  dominicks   133  10.022514     0   1.64  0.209602  0.528362   
28920    137  dominicks   134   9.086137     0   1.99  0.209602  0.528362   
28921    137  dominicks   135   9.086137     0   1.47  0.209602  0.528362   
28922    137  dominicks   136  11.390170     1   1.29  0.209602  0.528362   
28923    137  dominicks   137  10.606189     0   1.35  0.209602  0.528362   
28924    137  dominicks   138   8.886271     0   1.99  0.209602  0.528362   
28925    137  dominicks   139   9.156095     0   2.01  0.209602  0.528362   
28926    137  dominicks   140   8.691483     0   2.09  0.209602  0.528362   
28927    137  dominicks   141  10.641990     0   1.69  0.209602  0.528362   
28928    137  dominicks   142  11.403111     0   1.68  0.209602  0.528362   
28929    137  dominicks   143  10.667652     0   1.74  0.209602  0.528362   
28930    137  dominicks   144   9.618469     0   2.09  0.209602  0.528362   
28931    137  dominicks   145   8.886271     0   2.04  0.209602  0.528362   
28932    137  dominicks   146   9.426741     0   1.98  0.209602  0.528362   
28933    137  dominicks   147   8.841014     0   1.74  0.209602  0.528362   
28934    137  dominicks   148  10.377483     0   1.32  0.209602  0.528362   
28935    137  dominicks   149   9.869310     0   1.50  0.209602  0.528362   
28936    137  dominicks   150   9.924074     0   1.69  0.209602  0.528362   
289

# 1.	Find the dimensions of the oj data set

In [17]:
oj.shape

(28947, 17)

# 2.	Find the structure of the data set


In [18]:
oj.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28947 entries, 0 to 28946
Data columns (total 17 columns):
store       28947 non-null int64
brand       28947 non-null object
week        28947 non-null int64
logmove     28947 non-null float64
feat        28947 non-null int64
price       28947 non-null float64
AGE60       28947 non-null float64
EDUC        28947 non-null float64
ETHNIC      28947 non-null float64
INCOME      28947 non-null float64
HHLARGE     28947 non-null float64
WORKWOM     28947 non-null float64
HVAL150     28947 non-null float64
SSTRDIST    28947 non-null float64
SSTRVOL     28947 non-null float64
CPDIST5     28947 non-null float64
CPWVOL5     28947 non-null float64
dtypes: float64(13), int64(3), object(1)
memory usage: 3.8+ MB


# 3.	Find out the column names in the data set

In [64]:
oj.columns.values


array(['store', 'brand', 'week', 'logmove', 'feat', 'price', 'AGE60',
       'EDUC', 'ETHNIC', 'INCOME', 'HHLARGE', 'WORKWOM', 'HVAL150',
       'SSTRDIST', 'SSTRVOL', 'CPDIST5', 'CPWVOL5', 'LogInc'],
      dtype=object)

# 4.	Describe the data set

In [20]:
oj.describe()

store          week       logmove          feat         price  \
count  28947.000000  28947.000000  28947.000000  28947.000000  28947.000000   
mean      80.883511    100.459944      9.167864      0.237261      2.282488   
std       35.576511     34.692314      1.019378      0.425411      0.648001   
min        2.000000     40.000000      4.158883      0.000000      0.520000   
25%       53.000000     70.000000      8.489616      0.000000      1.790000   
50%       86.000000    101.000000      9.034080      0.000000      2.170000   
75%      111.000000    130.000000      9.764685      0.000000      2.730000   
max      137.000000    160.000000     13.482016      1.000000      3.870000   

              AGE60          EDUC        ETHNIC        INCOME       HHLARGE  \
count  28947.000000  28947.000000  28947.000000  28947.000000  28947.000000   
mean       0.173130      0.225220      0.155557     10.616735      0.115602   
std        0.061872      0.109945      0.187581      0.282314      0.030168   
min        0.058054      0.049550      0.024247      9.867083      0.013506   
25%        0.122100      0.145985      0.041910     10.456079      0.097938   
50%        0.170655      0.229390      0.074656     10.635326      0.111221   
75%        0.213949      0.284395      0.187761     10.796960      0.135168   
max        0.307398      0.528362      0.995691     11.236197      0.216354   

            WORKWOM       HVAL150      SSTRDIST       SSTRVOL       CPDIST5  \
count  28947.000000  28947.000000  28947.000000  28947.000000  28947.000000   
mean       0.359178      0.343766      5.097274      1.207317      2.120359   
std        0.052673      0.239028      3.472386      0.526528      0.729828   
min        0.244463      0.002509      0.132097      0.400000      0.772530   
25%        0.312636      0.123486      2.767046      0.727273      1.626192   
50%        0.355635      0.346154      4.650687      1.115385      1.963412   
75%        0.402313      0.528313      6.650602      1.538462      2.533672   
max        0.472308      0.916699     17.855951      2.571429      4.107902   

            CPWVOL5  
count  28947.000000  
mean       0.438914  
std        0.219248  
min        0.094562  
25%        0.271673  
50%        0.383227  
75%        0.560240  
max        1.143367

# 1.	Fetch the first row 3rd column from the data set

In [21]:
oj.iloc[0,2]

40

# 2.	Fetch the first, second and Third columns of the oj data frame

In [63]:
oj.iloc[:,0:3].head(5)

store      brand  week
0      2  tropicana    40
1      2  tropicana    46
2      2  tropicana    47
3      2  tropicana    48
4      2  tropicana    50

# 3.	Fetch the first, second, eighth and the 456th rows of the 1st, third and the sixth columns of the data frame

In [60]:
oj.loc[[0,1,7,455],['store','week','price']]

store  week  price
0        2    40   3.87
1        2    46   3.87
7        2    53   3.29
455      5    51   2.99

# 4.	Fetch the top 5 rows of the brand column

In [59]:
oj.iloc[0:5,1]

0    tropicana
1    tropicana
2    tropicana
3    tropicana
4    tropicana
Name: brand, dtype: object

# 5.	Fetch top 5 rows of the brand, week and feat details

In [57]:
oj.ix[[0,1,2,3,4],['brand','week','feat']]

brand  week  feat
0  tropicana    40     0
1  tropicana    46     0
2  tropicana    47     0
3  tropicana    48     0
4  tropicana    50     0

# 6.	Fetch the details of all distinct stores

In [56]:
oj.groupby(['store']).describe().head(10)

AGE60                                                                  \
       count      mean           std       min       25%       50%       75%   
store                                                                          
2      330.0  0.232865  1.723459e-15  0.232865  0.232865  0.232865  0.232865   
5      348.0  0.117368  9.867417e-16  0.117368  0.117368  0.117368  0.117368   
8      354.0  0.252394  5.558972e-16  0.252394  0.252394  0.252394  0.252394   
9      351.0  0.269119  1.222989e-15  0.269119  0.269119  0.269119  0.269119   
12     345.0  0.178341  9.172643e-16  0.178341  0.178341  0.178341  0.178341   

                CPDIST5            ...     price         week              \
            max   count      mean  ...       75%   max  count        mean   
store                              ...                                      
2      0.232865   330.0  1.927280  ...    2.8375  3.87  330.0  103.872727   
5      0.117368   348.0  1.600573  ...    2.9400  3.66  348.0  100.456897   
8      0.252394   354.0  2.905384  ...    2.6200  3.19  354.0   99.796610   
9      0.269119   351.0  1.820474  ...    2.6750  3.39  351.0   99.444444   
12     0.178341   345.0  0.839303  ...    2.9900  3.87  345.0  100.417391   

                                                     
             std   min    25%    50%     75%    max  
store                                                
2      33.508002  40.0  75.00  105.5  133.00  160.0  
5      34.657083  40.0  70.75   99.5  130.25  160.0  
8      35.148707  40.0  69.00  100.5  130.00  160.0  
9      35.035030  40.0  69.00  100.0  129.00  160.0  
12     35.296662  41.0  69.00  100.0  132.00  160.0  

[5 rows x 128 columns]

# 7.	Fetch all the observations for Tropicana brand

In [27]:
print(oj[oj['brand'] == 'tropicana'])

       store      brand  week    logmove  feat  price     AGE60      EDUC  \
0          2  tropicana    40   9.018695     0   3.87  0.232865  0.248935   
1          2  tropicana    46   8.723231     0   3.87  0.232865  0.248935   
2          2  tropicana    47   8.253228     0   3.87  0.232865  0.248935   
3          2  tropicana    48   8.987197     0   3.87  0.232865  0.248935   
4          2  tropicana    50   9.093357     0   3.87  0.232865  0.248935   
5          2  tropicana    51   8.877382     0   3.87  0.232865  0.248935   
6          2  tropicana    52   9.294682     0   3.29  0.232865  0.248935   
7          2  tropicana    53   8.954674     0   3.29  0.232865  0.248935   
8          2  tropicana    54   9.049232     0   3.29  0.232865  0.248935   
9          2  tropicana    57   8.613230     0   3.29  0.232865  0.248935   
10         2  tropicana    58   8.680672     0   3.56  0.232865  0.248935   
11         2  tropicana    59   9.034080     0   3.56  0.232865  0.248935   

# 8.	Fetch all the observations for Tropicana brand using query function

In [52]:
oj.query("(brand=='tropicana')").head(5)

store      brand  week   logmove  feat  price     AGE60      EDUC   ETHNIC  \
0      2  tropicana    40  9.018695     0   3.87  0.232865  0.248935  0.11428   
1      2  tropicana    46  8.723231     0   3.87  0.232865  0.248935  0.11428   
2      2  tropicana    47  8.253228     0   3.87  0.232865  0.248935  0.11428   
3      2  tropicana    48  8.987197     0   3.87  0.232865  0.248935  0.11428   
4      2  tropicana    50  9.093357     0   3.87  0.232865  0.248935  0.11428   

      INCOME   HHLARGE   WORKWOM   HVAL150  SSTRDIST   SSTRVOL  CPDIST5  \
0  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
1  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
2  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
3  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
4  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   

    CPWVOL5   LogInc  
0  0.376927  2.35643  
1  0.376927  2.35643  
2  0.376927  2.35643  
3  0.376927  2.35643  
4  0.376927  2.35643

# 9.	Fetch bottom 5 observations for those who have bought Tropicana or dominics

In [51]:
oj.query("(brand=='tropicana')|(brand=='dominicks')").tail(5)

store      brand  week    logmove  feat  price     AGE60      EDUC  \
28942    137  dominicks   156   7.987524     0   2.00  0.209602  0.528362   
28943    137  dominicks   157   8.502689     0   1.94  0.209602  0.528362   
28944    137  dominicks   158   9.505991     0   1.59  0.209602  0.528362   
28945    137  dominicks   159  10.927376     0   1.49  0.209602  0.528362   
28946    137  dominicks   160   8.859363     0   1.75  0.209602  0.528362   

        ETHNIC    INCOME   HHLARGE   WORKWOM   HVAL150  SSTRDIST   SSTRVOL  \
28942  0.11325  10.96649  0.092996  0.330293  0.860739  6.026484  0.705882   
28943  0.11325  10.96649  0.092996  0.330293  0.860739  6.026484  0.705882   
28944  0.11325  10.96649  0.092996  0.330293  0.860739  6.026484  0.705882   
28945  0.11325  10.96649  0.092996  0.330293  0.860739  6.026484  0.705882   
28946  0.11325  10.96649  0.092996  0.330293  0.860739  6.026484  0.705882   

       CPDIST5   CPWVOL5    LogInc  
28942  0.77253  0.333761  2.394844  
28943  0.77253  0.333761  2.394844  
28944  0.77253  0.333761  2.394844  
28945  0.77253  0.333761  2.394844  
28946  0.77253  0.333761  2.394844

# 10.	Fetch the income, brand, price observations with Tropicana brand without feature advertisement

In [30]:
oj.query("(brand == 'tropicana')&(feat == 0)")[["INCOME","brand","price"]]

INCOME      brand  price
0      10.553205  tropicana   3.87
1      10.553205  tropicana   3.87
2      10.553205  tropicana   3.87
3      10.553205  tropicana   3.87
4      10.553205  tropicana   3.87
5      10.553205  tropicana   3.87
6      10.553205  tropicana   3.29
7      10.553205  tropicana   3.29
8      10.553205  tropicana   3.29
9      10.553205  tropicana   3.29
10     10.553205  tropicana   3.56
11     10.553205  tropicana   3.56
12     10.553205  tropicana   3.56
13     10.553205  tropicana   3.56
14     10.553205  tropicana   3.87
15     10.553205  tropicana   2.99
16     10.553205  tropicana   2.99
17     10.553205  tropicana   3.59
18     10.553205  tropicana   3.59
19     10.553205  tropicana   2.39
20     10.553205  tropicana   2.39
21     10.553205  tropicana   3.59
22     10.553205  tropicana   2.59
23     10.553205  tropicana   2.59
24     10.553205  tropicana   3.59
25     10.553205  tropicana   3.59
26     10.553205  tropicana   2.49
27     10.553205  tropicana   3.59
28     10.553205  tropicana   3.59
29     10.553205  tropicana   3.59
...          ...        ...    ...
28710  10.966490  tropicana   2.39
28711  10.966490  tropicana   2.34
28712  10.966490  tropicana   2.29
28713  10.966490  tropicana   2.37
28714  10.966490  tropicana   3.18
28715  10.966490  tropicana   3.19
28716  10.966490  tropicana   3.19
28718  10.966490  tropicana   3.19
28720  10.966490  tropicana   1.79
28721  10.966490  tropicana   1.79
28722  10.966490  tropicana   1.95
28723  10.966490  tropicana   2.76
28725  10.966490  tropicana   2.59
28726  10.966490  tropicana   3.19
28727  10.966490  tropicana   2.74
28729  10.966490  tropicana   3.19
28730  10.966490  tropicana   3.19
28731  10.966490  tropicana   3.19
28732  10.966490  tropicana   3.19
28734  10.966490  tropicana   2.25
28737  10.966490  tropicana   2.67
28738  10.966490  tropicana   3.15
28739  10.966490  tropicana   3.18
28740  10.966490  tropicana   3.19
28741  10.966490  tropicana   3.17
28743  10.966490  tropicana   3.19
28745  10.966490  tropicana   2.77
28747  10.966490  tropicana   2.66
28748  10.966490  tropicana   3.11
28750  10.966490  tropicana   2.88

[8045 rows x 3 columns]

# 11.	Add a new column in the dataset: logInc which is the logarithm of the income

In [50]:
#oj['LogInc']=list(map(lambda x: math.log(x),o'j.INCOME))
oj['LogInc'] = np.log(oj.INCOME)
oj.head()

store      brand  week   logmove  feat  price     AGE60      EDUC   ETHNIC  \
0      2  tropicana    40  9.018695     0   3.87  0.232865  0.248935  0.11428   
1      2  tropicana    46  8.723231     0   3.87  0.232865  0.248935  0.11428   
2      2  tropicana    47  8.253228     0   3.87  0.232865  0.248935  0.11428   
3      2  tropicana    48  8.987197     0   3.87  0.232865  0.248935  0.11428   
4      2  tropicana    50  9.093357     0   3.87  0.232865  0.248935  0.11428   

      INCOME   HHLARGE   WORKWOM   HVAL150  SSTRDIST   SSTRVOL  CPDIST5  \
0  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
1  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
2  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
3  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   
4  10.553205  0.103953  0.303585  0.463887  2.110122  1.142857  1.92728   

    CPWVOL5   LogInc  
0  0.376927  2.35643  
1  0.376927  2.35643  
2  0.376927  2.35643  
3  0.376927  2.35643  
4  0.376927  2.35643

# 12.	Sort the Data in the increasing order of the week

In [49]:
sorted(oj['week'])

[40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,
 40,


# 13.	Sort the data in the decreasing order of Income

In [48]:
sorted(oj.INCOME,reverse = True)

[11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23619652,
 11.23

# 14.	Find the mean of the juice price for each brand

In [34]:
oj.groupby('brand').agg({'price':np.mean})

price
brand                
dominicks    1.735809
minute.maid  2.241162
tropicana    2.870493

# 15.	Find the average income for each brand and at each store

In [35]:
oj.groupby(["store","brand"]).agg({'INCOME':np.mean})

INCOME
store brand                 
2     dominicks    10.553205
      minute.maid  10.553205
      tropicana    10.553205
5     dominicks    10.922371
      minute.maid  10.922371
      tropicana    10.922371
8     dominicks    10.597010
      minute.maid  10.597010
      tropicana    10.597010
9     dominicks    10.787152
      minute.maid  10.787152
      tropicana    10.787152
12    dominicks     9.996659
      minute.maid   9.996659
      tropicana     9.996659
14    dominicks    11.043929
      minute.maid  11.043929
      tropicana    11.043929
18    dominicks    10.391976
      minute.maid  10.391976
      tropicana    10.391976
21    dominicks    10.716194
      minute.maid  10.716194
      tropicana    10.716194
28    dominicks    10.798534
      minute.maid  10.798534
      tropicana    10.798534
32    dominicks    10.674475
      minute.maid  10.674475
      tropicana    10.674475
...                      ...
123   dominicks    10.334100
      minute.maid  10.334100
      tropicana    10.334100
124   dominicks    10.258957
      minute.maid  10.258957
      tropicana    10.258957
126   dominicks    10.980876
      minute.maid  10.980876
      tropicana    10.980876
128   dominicks    10.153429
      minute.maid  10.153429
      tropicana    10.153429
129   dominicks    10.999864
      minute.maid  10.999864
      tropicana    10.999864
130   dominicks     9.966650
      minute.maid   9.966650
      tropicana     9.966650
131   dominicks    10.793537
      minute.maid  10.793537
      tropicana    10.793537
132   dominicks    10.619154
      minute.maid  10.619154
      tropicana    10.619154
134   dominicks    10.722518
      minute.maid  10.722518
      tropicana    10.722518
137   dominicks    10.966490
      minute.maid  10.966490
      tropicana    10.966490

[249 rows x 1 columns]

# 16.	Find:a.	Mean and std deviation of the income


In [47]:
print(oj.agg({'INCOME':np.mean}))

print(oj.agg({'INCOME':np.std}))

#   (or)
#arr = oj['INCOME']
#arr

#np.std(arr,axis=0)

INCOME    10.616735
dtype: float64
INCOME    0.282314
dtype: float64


# b.	For income greater than or equal to 10.5, find the mean income

In [45]:
print(oj.agg({'INCOME':np.mean}))

INCOME    10.616735
dtype: float64


# c.	For each brand having price >=2.5 find the mean, median, sd of the log of income

In [43]:
a = oj[oj.price >= 2.5]
#a.groupby('brand').agg({'LogInc':np.mean})
#a.groupby('brand').agg({'LogInc':np.median})
#a.groupby('brand').agg({'LogInc':np.std})

 

    
a.groupby('brand')[['LogInc']].agg(['mean','std','median'])


LogInc                    
                 mean       std    median
brand                                    
dominicks    2.348104  0.038822  2.338646
minute.maid  2.360556  0.028522  2.363903
tropicana    2.361968  0.026782  2.364181

# 17.	Find the Cross tabulation of brands and feature advertisement

In [39]:
#oj.groupby('brand').agg('brand','feat')
#oj.pivot_table(columns=['brand','feat'])
pd.crosstab(oj.brand,oj.feat)

feat            0     1
brand                  
dominicks    7169  2480
minute.maid  6865  2784
tropicana    8045  1604